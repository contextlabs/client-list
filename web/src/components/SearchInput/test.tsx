import { render, screen } from '@testing-library/react';

import { SearchInput } from '.';

const noop = () => {};

describe('<SearchInput />', () => {
  const props = {
    setSearchTerm: noop
  };

  it('should render the SearchInput as default', () => {
    const { container } = render(<SearchInput {...props} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByRole('textbox', { name: /SearchInput/i })).toBeInTheDocument();
    expect(screen.getByRole('button', { name: /SearchInputIcon/i })).toBeInTheDocument();
  });
});
