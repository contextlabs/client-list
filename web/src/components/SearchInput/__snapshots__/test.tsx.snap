// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchInput /> should render the SearchInput as default 1`] = `
<div
  class="flex items-center justify-center"
>
  <div
    class="flex border-2 rounded w-full"
  >
    <input
      aria-label="SearchInput"
      class="px-4 py-2 w-full"
      placeholder="Search..."
      type="text"
    />
    <button
      aria-label="SearchInputIcon"
      class="flex items-center justify-center px-4 border-l bg-white"
    >
      <div
        class="w-6 h-6 text-gray-600"
      />
    </button>
  </div>
</div>
`;
